network-structure.md

```diff
@@ -0,0 +1,36 @@
+# Network structure
+
+## Message Layer
+The Message Layer is the communication layer that concerns itself with (de)composition, validation, serialisation, etc. of ECP Messages (see {ref}`messages.md#messages`).
+
+## Transport Layer
+The Transport Layer is the communication layer that transports ECP Messages between Components, making use of routing information in the Message header.
+This uses zeromq or simpler localised methods, see the section "Message Transport Mode".
+
+## Node
+A Node is a local context in which (part of) an ECP deployment runs. 
+This may be a single application using one or more threads or processes. 
+
+An ECP network has one or more Nodes.
+Components within a Node may use the Local Message Transport (LMT/"local mode") or the Distributed Message Transport (DMT/"distributed mode") to communicate with each other, see below.
+All Components of a Node must use the same Message Transport Mode.
+:::{admonition} Note
+In the future this might be relaxed, so that Components within a Node might operate in different modes.
+:::
+:::{admonition} TODO
+Decide how we track which Node a Component belongs to, and how the mode information propagates within a Node.
+:::
+
+### Bridging Coordinators
```

Comment:
Let us move this part out of a node, maybe a general comment is better.
There we can state, that every Component (except for Coordinators) is connected to exactly one Coordinator. The Coordinators are connected to each other, such that any Component may send a message to any other Component via their respective Coordinators.

Reply:
Good idea, I modified this a bit (all your relationships are actually one-on-one, a flowchart offers the possibiltiy "Node" subgraphs.
```mermaid
flowchart LR

    subgraph Node1
        Component1 -- "connects to" --- Coordinator1
        Component2 -- "connects to" --- Coordinator1
    end
    Coordinator1 -- "coordinates with" --- Coordinator2
    subgraph Node2
        Coordinator2 -- "connects to" --- Component3
        Coordinator2 -- "connects to" --- Component4
    end
```

Reply:
Nice.